addons/mail/static/src/model/model_manager.js

```diff
@@ -993,30 +984,36 @@ export class ModelManager {
         const allNames = Object.keys(registry);
         const Models = {};
         const generatedNames = [];
-        let toGenerateNames = [...allNames];
-        while (toGenerateNames.length > 0) {
-            const generatable = toGenerateNames.map(name => registry[name]).find(entry => {
-                let isGenerateable = true;
-                for (const dependencyName of entry.dependencies) {
-                    if (!generatedNames.includes(dependencyName)) {
-                        isGenerateable = false;
-                    }
+        for (const modelName of ['mail.model', ...allNames.filter(name => name !== 'mail.model')]) {
+            const definition = registry[modelName];
+            // Create the model and give it a meaningful name to be displayed in
+            // stack traces and stuff.
+            const Model = modelName === 'mail.model'
+                ? { [modelName]: class {} }[modelName]
+                : { [modelName]: class extends Models['mail.model'] {} }[modelName];
+            if (modelName === 'mail.model') {
+                for (const [getterName, getter] of Object.entries(definition.modelGetters)) {
+                    Object.defineProperty(Model, getterName, { get: getter });
+                }
+                for (const [getterName, getter] of Object.entries(definition.recordGetters)) {
+                    Object.defineProperty(Model.prototype, getterName, { get: getter });
```

Comment:
Big expression for making an iterable collection + big ternary conditional expression + `if` that occurs only once.
I think it's possible to make some improvements so it's more readable. Like:

- have a statement solely for building the ordered collection
- isolate LOCs in (properly named) function
- make 2 groups of statements, the former to explicitly create `mail.model`, and the latter for all models.

These are just ideas. Not sure whether readability will be much improved, but right now it's not that pretty, so could be worth testing some other ways to implement to improve readability.